Source/CarthageKit/Project.swift

```diff
@@ -531,6 +572,31 @@ public final class Project {
 			}
 	}
 
+	/// Emits the framework URL if it matches the local swift version and `empty` if not.
+	internal func matchingSwiftVersionURL(_ frameworkURL: URL) -> SignalProducer<URL, CarthageError> {
+		return SignalProducer.combineLatest(swiftVersion, frameworkSwiftVersion(frameworkURL))
+			.flatMap(.merge) { swiftVersion, frameworkVersion in
+				return swiftVersion == frameworkVersion
+					? SignalProducer(value: frameworkURL)
+					: SignalProducer(error: .incompatibleFrameworkSwiftVersions(local: swiftVersion, framework: frameworkVersion))
+			}
+			.take(first: 1)
```

Comment:
Sorry, that's an old habit when doing a `combineLatest` and intending for the result to be a `Single` since if anything changes with the signals you are observing you could easily wind up producing more than a single value. Because that would mean in this case multiple copies of a framework in this case, I thought I'd be safe. I will remove it.